trestle/tasks/xlsx_to_oscal_profile.py

```diff
@@ -0,0 +1,180 @@
+
+        Args:
+            config_object: Config section associated with the task.
+        """
+        super().__init__(config_object)
+        self.xlsx_helper = XlsxHelper()
+        self._timestamp = datetime.datetime.utcnow().replace(microsecond=0).replace(tzinfo=datetime.timezone.utc
+                                                                                    ).isoformat()
+
+    def set_timestamp(self, timestamp: str) -> None:
+        """Set the timestamp."""
+        self._timestamp = timestamp
+
+    def print_info(self) -> None:
+        """Print the help string."""
+        self.xlsx_helper.print_info(self.name, 'profile')
+
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        return TaskOutcome('simulated-success')
+
+    def execute(self) -> TaskOutcome:
+        """Provide an executed outcome."""
+        try:
+            return self._execute()
+        except Exception:
+            logger.info(traceback.format_exc())
+            return TaskOutcome('failure')
+
+    def _execute(self) -> TaskOutcome:
+        """Execute path core."""
+        if not self.xlsx_helper.configure(self):
+            return TaskOutcome('failure')
+        # config output
+        odir = self._config.get('output-dir')
+        opth = pathlib.Path(odir)
+        self._overwrite = self._config.getboolean('output-overwrite', True)
+        # insure output dir exists
+        opth.mkdir(exist_ok=True, parents=True)
+        # calculate output file name & check writability
+        oname = 'profile.json'
+        ofile = opth / oname
+        if not self._overwrite and pathlib.Path(ofile).exists():
+            logger.error(f'output: {ofile} already exists')
+            return TaskOutcome('failure')
+        # create OSCAL Profile
+        metadata = Metadata(
+            title='Profile for ' + self._get_profile_title(),
+            last_modified=self._timestamp,
+            oscal_version=OSCAL_VERSION,
+            version=get_trestle_version(),
+        )
+        import_ = Import(
+            href=self._get_spread_sheet_url(),
+            include_controls=self._get_include_controls(),
+        )
+        imports = [import_]
+        set_parameters = self._get_set_parameters()
+        modify = Modify(set_parameters=set_parameters)
+        profile = Profile(
+            uuid=str(uuid.uuid4()),
+            metadata=metadata,
+            imports=imports,
+            modify=modify,
+        )
+        # write OSCAL Profile to file
+        if self._verbose:
+            logger.info(f'output: {ofile}')
+        profile.oscal_write(pathlib.Path(ofile))
+        # issues
+        self._report_issues()
+        return TaskOutcome('success')
+
+    def _get_include_controls(self) -> List[SelectControlById]:
+        """Get include controls from spread sheet."""
+        include_control = SelectControlById(with_ids=self._get_with_ids())
+        return [include_control]
```

Comment:
Do you want to specify a with_child_controls value here?

Reply:
For now, no. It is not required.